src/o365/graph/commands/teams/teams-add.ts

```diff
@@ -0,0 +1,197 @@
+        var teamsId = this.GetTeamsIdFromResponse(res, args);
+        cmd.log({Team:teamsId});
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+        cb();
+      }, (err: any): void =>{ 
+        this.handleRejectedODataJsonPromise(err, cmd, cb)
+      });
+  }
+
+  private GetTeamsIdFromResponse(res:any, args: CommandArgs) : string {
+    let teamsRspHdrRegEx;
+    if(!args.options.groupId) {
+      teamsRspHdrRegEx = /(teams\(')([0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12})('\))/i.exec(res.headers.location);
+    }
+    else {
+      teamsRspHdrRegEx = /(team\(')([0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12})('\))/i.exec(res.headers.location);
+    }
+
+    if(teamsRspHdrRegEx != null && teamsRspHdrRegEx.length == 4)
+    {
+      return teamsRspHdrRegEx[2];
+    }
+
+    return '';
+  }
+
+  private CreateTeamRequest(cmd : CommandInstance, args: CommandArgs) : request.RequestPromise {
+    const teamsEndpoint: string = `${auth.service.resource}/beta/teams`;
+    const teamsRequestBody = {
+      'template@odata.bind': 'https://graph.microsoft.com/beta/teamsTemplates/standard',
+      displayName: args.options.name,
+      description: args.options.description
+    };
+
+
+    const requestOptions: any = {
+      url: teamsEndpoint,
+      resolveWithFullResponse: true ,
+      headers: Utils.getRequestHeaders({
+        authorization: `Bearer ${auth.service.accessToken}`,
+        accept: 'application/json;odata.metadata=none',
+        'content-type': 'application/json;odata=nometadata'
+      }),
+      body: teamsRequestBody,
+      json: true
+    };
+
+    if (this.debug) {
+      cmd.log('Executing web request...');
+      cmd.log(requestOptions);
+      cmd.log('');
+    }
+
+    return request.post(requestOptions);
+  }
+
+  private CreateGroupTeamRequest(cmd : CommandInstance, args: CommandArgs) : request.RequestPromise {
+    const groupTeamsEndpoint: string = `${auth.service.resource}/beta/groups/${args.options.groupId}/team`;
+   
+    const requestOptions: any = {
+      url: groupTeamsEndpoint,
+      resolveWithFullResponse: true ,
+      headers: Utils.getRequestHeaders({
+        authorization: `Bearer ${auth.service.accessToken}`,
+        accept: 'application/json;odata.metadata=none',
+        'content-type': 'application/json;odata=nometadata'
+      }),
+      body: {},
+      json: true
+    };
+
+    if (this.debug) {
+      cmd.log('Executing web request...');
+      cmd.log(requestOptions);
+      cmd.log('');
+    }
+
+    return request.put(requestOptions);
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-n, --name [name]',
+        description: 'Display name for the Microsoft Teams team'
+      },
+      {
+        option: '-d, --description [description]',
+        description: 'Description for the Microsoft Teams team'
```

Comment:
We should clarify that description is required when groupId is not specified

Reply:
Description for the 'name' parameter amended to specify that it is not required with the 'groupId'.